cub/cub/util_type.cuh

```diff
@@ -209,14 +209,23 @@ struct NullType
  * dispatch based on constant integral values)
  */
 template <int A>
-struct Int2Type
+struct CCCL_DEPRECATED_BECAUSE("Use ::cuda::std::integral_constant instead") Int2Type
 {
   enum
   {
     VALUE = A
   };
 };
 
+template <int Value>
+using int_constant_t = ::cuda::std::integral_constant<int, Value>;
+
+template <auto Value>
+inline constexpr auto int_constant_v = int_constant_t<Value>{};
+
+template <bool Value>
+inline constexpr auto bool_constant_v = ::cuda::std::bool_constant<Value>{};
+
```

Comment:
Important: Let's put the new helpers inside a `detail` namespace, so we can change them more flexibly in the future.